### ISSUE TITLE
Behavior change in 1.1.178: cython class infered as module

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#2446]
**Describe the bug**
When running pyright 1.1.178+ on the pandas code base (pandas is currently using 1.1.171), pyright infers `pd.Interval` as `Module("pandas.core.indexes.interval.Interval")`, version 1.1.77 infers it as `"Type[Float64Index] | Type[Int64Index] | Type[UInt64Index] | Type[datetime] | Module("numpy") | Any | Type[SparseArray]"`.

Based on https://github.com/pandas-dev/pandas/blob/0b7081840ce034e16ce22fc001c44ac50bafc0c9/pandas/_libs/interval.pyx#L215 `pd.Interval` should be a class.

**To Reproduce**
```sh
$ git clone --depth=1 -b master git@github.com:pandas-dev/pandas.git
$ cd pandas
$ echo "reportMissingParameterType = false" >> pyproject.toml
$ python setup.py build_ext -j 4
$ pyright
pandas/_typing.py:90:59 - error: Module cannot be used as a type
pandas/_typing.py:117:25 - error: Module cannot be used as a type
pandas/core/arrays/interval.py:97:22 - error: Module cannot be used as a type
pandas/core/arrays/period.py:1155:24 - error: Module cannot be used as a type

```

Probably related to this changelog entry of 1.1.178:

> Bug Fix: Fixed false positive error that occurred when importing a symbol in a from x import y statement that referred to a chain of imports and was eventually resolved to a native library (e.g. ".pyd" or ".so" file).

**Comment by erictraut** [comment]
Ooh, sorry about the regression. Thanks for reporting it. You are correct that the regression was introduced as part of that other bug fix.

This will be fixed in the next release.
